M2-6103: Updated Empty States

JustinNusca opened the pull request:
### 📝 Description

🔗 [M2-6103](https://mindlogger.atlassian.net/browse/M2-6103): [FE] [Participants Overview] List View - Empty State

This PR updates the Empty State for a given applet's "Participants" and "Respondents" tables to match the design in Figma. 

This change modifies the Empty State's icon for all instances of the `DashboardTable`, which also includes a couple other uses (invitations & managers), but I think this is acceptable — no designs for these empty states are provided in Figma that would contradict this, and all instances of the DashboardTable are used to represent rows of users, which is what the icon communicates.

As part of this change, I also modified how `svgSprite` was constructed, in order to allow for more specific prop type validation for SVG icons, like so:

![Screenshot 2024-04-09 at 2 33 39 PM](https://github.com/ChildMindInstitute/mindlogger-admin/assets/1670836/913e611e-838a-4674-9473-bb227848def1)

This required updating proptypes on some unrelated components, and in some cases fixing some incorrect icon IDs that were exposed by this change, but there shouldn't be any in-app difference.

### 📸 Screenshots

| Before | After |
|-|-|
| ![localhost_3000_dashboard_76112bef-c9f9-4c72-8165-5bc0b64ea609_participants(iPad Pro) (1)](https://github.com/ChildMindInstitute/mindlogger-admin/assets/1670836/7eac5f9a-acce-40e0-9db8-d743507a6ed0) | ![localhost_3000_dashboard_76112bef-c9f9-4c72-8165-5bc0b64ea609_respondents(iPad Pro)](https://github.com/ChildMindInstitute/mindlogger-admin/assets/1670836/acd35917-ccb8-4605-9ea0-ce83b827c0a6) |
| ![localhost_3000_dashboard_76112bef-c9f9-4c72-8165-5bc0b64ea609_participants(iPad Pro)](https://github.com/ChildMindInstitute/mindlogger-admin/assets/1670836/b28799cc-869a-430b-894c-c6d1b30d2127) | ![localhost_3000_dashboard_76112bef-c9f9-4c72-8165-5bc0b64ea609_respondents(iPad Pro) (1)](https://github.com/ChildMindInstitute/mindlogger-admin/assets/1670836/ae782855-a6e5-447e-8c66-6c408f95ef12) |


[M2-6103]: https://mindlogger.atlassian.net/browse/M2-6103?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

JustinNusca committed:
M2-6103: Add better types for Icon component.

JustinNusca committed:
M2-6103: Update EmptyState component.

JustinNusca committed:
M2-6103: Update renderable content for Participant & Respondent empty states.

IhorNesterenko99 commented on src/modules/Dashboard/features/Respondents/Respondents.tsx:
```diff
@@ -415,7 +416,28 @@ export const Respondents = () => {
+      return appletId ? (
+        <Box
+          component="span"
+          sx={{
```
In cases where the sx prop contains more than three properties, it would be beneficial to consider using styled components. Please check other occurrences as well.

IhorNesterenko99 commented on src/modules/Dashboard/features/Respondents/Respondents.tsx:
```diff
@@ -415,7 +416,28 @@ export const Respondents = () => {
       }
 
-      return appletId ? t('noRespondentsForApplet') : t('noRespondents');
+      return appletId ? (
```
Could you please move the renderEmptyComponent function to its own component, such as Respondents/EmptyState? This could help separate concerns and improve code organization.

IhorNesterenko99 commented on src/modules/Dashboard/features/Participants/Participants.tsx:
```diff
@@ -469,7 +469,27 @@ export const Participants = () => {
       }
 
-      return appletId ? t('noParticipantsForApplet') : t('noParticipants');
+      return appletId ? (
```
Could you please move the renderEmptyComponent function to its own component, such as Participants/EmptyState?

JustinNusca replied:
Looks like you've already approved, but for posterity while I mark these as resolved: [This change has been made, here](https://github.com/ChildMindInstitute/mindlogger-admin/blob/9b4952bb45141cf866e41a4c415a86c54ab9893b/src/modules/Dashboard/components/EmptyDashboardTable/EmptyDashboardTable.tsx).

IhorNesterenko99 commented on src/svgSprite.ts:
```diff
@@ -1098,26 +1280,41 @@ export const svgSprite = (): string => `
+export type Icons = keyof typeof icons;
+const iconKeys = Object.keys(icons) as Icons[];
+
+export const svgSprite = (): string => `
```
What do you think about replacing the current implementation of svgSprite with the following code?
``export const svgSprite = (): string => {``
  ``const iconDefs = Object.values(icons).join('');``
  
   ``return `<svg><defs>${iconDefs}</defs></svg>`;``
``};``
This could slightly improve performance.

mbanting replied:
@IhorNesterenko99 not sure which would be more performant although the current version is slightly more readable.

How about this:
```
export const svgSprite = (): string => {
const iconDefs = Object.values(icons).join('');

return `
<svg>
  <defs>
    ${iconDefs}
  </defs>
</svg>
`;
};
```

IhorNesterenko99 replied:
@mbanting  It looks like your last code snippet aligns with my proposal. If there are no objections, we could implement it.

JustinNusca committed:
M2-6103: Update usage of icons to satisfy `Icons` type.

JustinNusca committed:
M2-6103: Address PR comments.

- Abstract empty state content to own component.
- Modify construction of svgSprite content.

mbanting reviewed (approved):
Nice work @JustinNusca

mbanting commented on src/resources/app-en.json:
```diff
@@ -802,8 +802,8 @@
-  "noRespondentsForApplet": "No Respondents yet. Invite your first one by clicking the ‘Add Respondent’ button above.",
-  "noParticipantsForApplet": "No Participants yet. Invite your first one by clicking the ‘Add Participant’ button above.",
+  "noRespondentsForApplet": "This applet has no respondents",
+  "noParticipantsForApplet": "This applet has no participants",
```
Please update the French version as well.

_Cette applet n'a aucun répondant_
_Cette applet n'a aucun participant_

(via Google translate)

JustinNusca replied:
[Updated](https://github.com/ChildMindInstitute/mindlogger-admin/pull/1645/commits/c1213d76cc2ae4f1f3ac7771596b13e8eeefc399)! 

In general, is maintaining the french content with machine-translated strings the ongoing practice, when there's copy changes?

mbanting replied:
For now, yes

JustinNusca committed:
M2-6103: Update french language json.

- Add corresponding updated copy from google-translate.
